internal/transports/p2p/peer/peer.go

```diff
@@ -555,6 +557,26 @@ func (p *Peer) handleHeadersMsg(msg *wire.MsgHeaders) {
 	}
 }
 
+func (p *Peer) handleGetHeadersMsg(msg *wire.MsgGetHeaders) {
+	p.log.Info().Msgf("received getheaders msg from peer %s", p)
+	if !p.syncedCheckpoints {
+		p.log.Info().Msgf("we are still syncing, ignoring getHeaders msg from peer %s", p)
+		return
+	}
+
+	locator := msg.BlockLocatorHashes
+	if len(locator) == 0 {
+		p.log.Info().Msgf("no locator hashes in getheaders msg from peer %s", p)
```

Comment:
maybe it's better to log.Warn

Reply:
changed to handle errors log at the peer side